### DISCUSSION TITLE
RollPass exceeds the maximum iteration count of 100 for all simulations with "pyroll-piller-model" .

### DISCUSSION THREAD

**Discussion by smkhalid** [pyroll-project/pyroll-core#255] [General]
Hi. I am running into an issue after installing the "pyroll-piller-model" library, i am having the following Log info message while running any simulation:

"Solution iteration of RollPass 'Stand A ' exceeded the maximum iteration count of 100. Continuing anyway."

I am working in Pycharm. After adding "pyroll-piller-model" to the python interpreator, i am having the issue throughout the project. Even when I do not import the "pyroll-piller-model" in a certain simulation, the same issue occurs.
The solution exceeds the maximum iteration of 100 for any specific Roll Pass Stand and never finishes.

I have tried it for several simulations in Projects but the problem remains. As soon as I uninstall the "pyroll-piller-model", the problem resolves.
The iteration-related issue is resolved when I turn off the Elongation Correction by "pyroll-piller-model. But it heavily impacts the deformation accuracy. So it cannot be a solution.

Has anyone else come across this, or does anyone have insights into what might be causing the iterations to get stuck? Any help or suggestions would be greatly appreciated.

Thanks in advance!

**Comment by sag-kalibreur** [comment]
Hello,
I have the same problem with the ‘pyroll-piller-model’ when calculating the upper roll pass for asymmetrical profiles:

`upper_pass_sequence.solve(in_profile)`

--> 'Solution iteration of RollPass 'Upper Roll Pass ' exceeded the maximum iteration count of 100. Continuing anyway.'

However, there is no effect on other notebooks and the results are correct in the end.

I work with Visual Studio Code including the Jupyter Notebook Plugin from Microsoft.
I haven't found out the cause of the problem yet. The problem also exists (of course) in the original notebook from Kalibreurstag, so maybe someone can take a look at it.

Have you tried restarting the Python kernel and/or clearing the cache before performing calculations in other projects/notebooks? It's possible that the pillar model remains loaded in memory if calculations were previously performed with it, causing problems.
I experience similar problems when running different notebooks or Python scripts one after the other. Restarting the kernel helps...

**Comment by RollingTechSupport** [comment]
Hi, 
So the problem was, that several spreading models loaded the `pillar-model`, this should now be fixed. 
If still have issues please file and Issue labeled as `Bug`.

**Reply by sag-kalibreur** [reply]
Well done Christoph, thanks for the fix...
